modules/network/src/main/java/org/apache/ignite/internal/network/serialization/ClassDescriptor.java

```diff
@@ -59,6 +65,20 @@
      */
     private final boolean isFinal;
 
+    /** Total number of bytes needed to store all primitive fields. */
+    private final int primitiveFieldsDataSize;
+    /** Total number of object (i.e. non-primitive) fields. */
+    private final int objectFieldsCount;
+
+    private Map<String, FieldDescriptor> fieldsByName;
+    /**
+     * Offsets into primitive fields data (which has size {@link #primitiveFieldsDataSize}).
+     * These are different from the offsets used in the context of {@link sun.misc.Unsafe}.
+     */
+    private Object2IntMap<String> primitiveFieldDataOffsets;
+    /** Indices of non-primitive (i.e. object) fields in the object fields array. */
```

Comment:
```suggestion
    /** Indices of non-primitive fields in the object fields array. */
```

Reply:
Same question as the one in the first suggestion here :)

Reply:
I think "non-primitive" equals "object", so no point in writing (i.e. object), no?

Reply:
Removed the snippet